packages/nc-gui/composables/useMultiSelect/index.ts

```diff
@@ -903,36 +912,75 @@ export function useMultiSelect(
           const rows = unref(data).slice(startRow, endRow + 1)
           const props = []
 
-          for (const row of rows) {
```

Comment:
refactored a bit. you may double check if `if (pasteValue === undefined) ` is really required or not. 

```ts
let pasteValue
for (const row of rows) {
  // TODO handle insert new row
  if (!row || row.rowMeta.new) continue

  for (const col of cols) {
    if (!col.title || !isPasteable(row, col)) {
      continue
    }

    if (e.clipboardData?.files?.length) {
      if (col.uidt !== UITypes.Attachment) {
        continue
      }

      if (pasteValue === undefined) {
        const fileUploadPayload = convertCellData(
          {
            value: '',
            files: e.clipboardData?.files,
            to: col.uidt as UITypes,
            column: col,
            appInfo: unref(appInfo),
          },
          isMysql(meta.value?.source_id),
          true,
        )

        if (fileUploadPayload?.length) {
          const uploadedFiles = await handleFileUpload(fileUploadPayload, col.id!)
          pasteValue = Array.isArray(uploadedFiles) && uploadedFiles.length ? JSON.stringify(uploadedFiles) : null
        }
      }
    } else {
      pasteValue = convertCellData(
        {
          value: clipboardData,
          to: col.uidt as UITypes,
          column: col,
          appInfo: unref(appInfo),
        },
        isMysql(meta.value?.source_id),
        true,
      )
    }

    props.push(col.title)

    if (pasteValue !== undefined) {
      row.row[col.title] = pasteValue
    }
  }
}
```

Reply:
I think `if (pasteValue === undefined)` is required as in some cases `convertCellData` function returning `undefined`. I will confirm this with @mertmit